CRM/Core/Payment.php

```diff
@@ -1155,20 +1155,20 @@ protected function getCurrency($params = []) {
   }
 
   /**
-   * Legacy. Better for a method to work on its own PropertyBag,
-   * but also, this function does not do very much.
+   * Get the submitted amount, padded to 2 decimal places, if needed.
    *
    * @param array $params
    *
    * @return string
-   * @throws \CRM_Core_Exception
    */
   protected function getAmount($params = []) {
     if (!CRM_Utils_Rule::numeric($params['amount'])) {
       CRM_Core_Error::deprecatedWarning('Passing Amount value that is not numeric is deprecated please report this in gitlab');
       return CRM_Utils_Money::formatUSLocaleNumericRounded(filter_var($params['amount'], FILTER_SANITIZE_NUMBER_FLOAT, FILTER_FLAG_ALLOW_FRACTION), 2);
     }
-    return CRM_Utils_Money::formatUSLocaleNumericRounded($params['amount'], 2);
+    // Amount is already formatted to a machine-friendly format but may NOT have
+    // decimal places - eg. it could be 1000.1 so this would return 1000.10.
+    return Civi::format()->machineMoney($params['amount'], 'USD', 'en_US');
```

Comment:
@eileenmcnaughton  This passes a third parameter (`en_US`); in the actual method, it doesn't take a third parameter, but it does hard-code a reference to `en_US`. 

I'm trying to understand the difference between "moneyNumber" and "machineNumber". Is the idea that "machineNumber" is supposed to be _less_ context-sensitive (ie the machineNumber format isn't actually about your specific locale; it's about some general norm that's seen most wire-protocols; and a way to match that norm is to use `en_US` with `GROUPING=>false`)?

Reply:
@totten so moneyNumber is a number formatted per the currency (but without the currency symbol) - eg. as you would do it in data entry

machineMoney is money formatted with the number of decimal places that the would be used for that currency but is machine-ready 

You are right the locale can go now - I changed it - the currency is only relevant in that it might (in future) be something like JPY in which case it would be 1000 not 1000.10

Reply:
OK, cool.

Flagging "merge ready" (ie any additional feedback from @stesi561 could sway it), but it's de-facto "merge on pass" (ie the time for 5.49 is at-hand; after another test-run, there's no specific reason to hold it up).